util/util.go

```diff
@@ -703,6 +703,91 @@ func ManagerDelegatingClientFunc(cache cache.Cache, config *rest.Config, options
 	}, nil
 }
 
+// DelegatingClientFuncWithUncached returns a manager.NewClientFunc to be used when creating
+// a new controller runtime manager.
+//
+// A delegating client reads from the cache and writes directly to the server.
+// This avoids getting unstructured objects directly from the server.
+//
+// In addition, this function builds a custom reader that used the input list of objects
+// to determine which GVK should always query the live server, instead of going through a cache.
+func DelegatingClientFuncWithUncached(uncachedObjects ...runtime.Object) func(cache cache.Cache, config *rest.Config, options client.Options) (client.Client, error) {
+	return func(cache cache.Cache, config *rest.Config, options client.Options) (client.Client, error) {
+		uncachedGVKs := make(map[schema.GroupVersionKind]struct{}, len(uncachedObjects))
+		for _, obj := range uncachedObjects {
+			gvk, err := apiutil.GVKForObject(obj, options.Scheme)
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to retrieve GVK for uncached object")
+			}
+			uncachedGVKs[gvk] = struct{}{}
+		}
+
+		// Create a live client.
+		c, err := client.New(config, options)
+		if err != nil {
+			return nil, err
+		}
+		return &client.DelegatingClient{
+			// The custom delegatingReader is defined in this package and is a custom client.Reader
+			// that routes the uncached GVK to the API Server using the live client created above.
+			//
+			// It also maintains the client.DelegatingReader behavior of bypassing the cache for
+			// unstructured objects and lists.
+			Reader: &delegatingReader{
+				liveReader:   c,
+				cacheReader:  cache,
+				uncachedGVKs: uncachedGVKs,
+				scheme:       options.Scheme,
+			},
+			Writer:       c,
+			StatusClient: c,
+		}, nil
+	}
+}
+
+type delegatingReader struct {
```

Comment:
the name initially confused me, because there are no hints about supporting live reads/uncached things, but I don't have better suggestions...
Also, what about moving this to CR (in a future iteration)?

Reply:
This is only needed for v0.3.x, it's ported back from Controller Runtime v0.7.x — given that the struct is unexported I don't see an issue in keeping this here only while we migrate to v0.4.0

Reply:
makes sense